share/completions/gsettings.fish

```diff
@@ -0,0 +1,99 @@
+# Fish completions for glibs's "gsettings" configuration tool
+
+function __fish_complete_gsettings_args
+    set -l schema_commands get monitor writable range describe set reset reset-recursively list-keys list-children list-recursively
+    set -l key_commands get monitor writable range describe set reset
+
+    set -l schemadir
+    set -l cmd (commandline -poc)
+    set -e cmd[1]
+
+    if set -q cmd[2]; and string match -q -- '--schemadir' $cmd[1]
```

Comment:
Are you sure that "--schemadir" can only appear before the command? Also, is it the only option that still allows commands?